### DISCUSSION TITLE
I can't call go method in js

### DISCUSSION THREAD

**Discussion by eric49861** [wailsapp/wails#2897] [Q&A]
go code:
```go
func (a *App) Greet(name string) string {
	return "Hello World, " + name
}
```
svelte:
```typescript
    function doGreet() {
        let name:string = "eric"
        Greet(name).then(result=>{
            console.log(result)
        })
    }
```
error:
![image](https://github.com/wailsapp/wails/assets/93255461/a3dcd023-a70d-4dd9-bbd6-da14a9339f5b)
App.js
![image](https://github.com/wailsapp/wails/assets/93255461/895de3ff-4650-461f-ae86-840f13969165)
this problem confuses me for a long time, i do the same thing as official guidance.

**Comment by eric49861** [comment]
when i print in console, i find that window['go'] is undefined

**Comment by ianmjones** [comment]
It's not shown in your Svelte code, but did you do something like the following at the top of your `<script>` tag? ...

```javascript
import {Greet} from "../wailsjs/go/main/App.js";
```

Also, are you seeing this when using `wails dev`?

If you're only using `wails build`, did you do a `wails generate module` before `wails build`?

**Reply by eric49861** [reply]
yeah，i have imported the package correctly.
I have solved this problem, because i debug the programme in a broswer, so i lost the environment of wails
